clcache.py

```diff
@@ -97,8 +97,8 @@ def childDirectories(path, absolute=True):
 def normalizeBaseDir(baseDir):
     if baseDir:
         baseDir = os.path.normcase(baseDir)
-        if not baseDir.endswith(os.path.sep):
-            baseDir += os.path.sep
+        if baseDir.endswith(os.path.sep):
+            baseDir = baseDir[0:-1]
```

Comment:
This was needed to make the integration test `TestBasedir.testBasedirIncludeArg` pass. There the include path is generated by `os.getcwd()` which doesn't include the trailing separator. As `normalizedBasedir` is used in `collapseBasedirToPlaceholder` the version with the trailing slash would not match.